scripts/files-touched-check.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import os
+import sys
+import subprocess
+
+travis_commit_range = os.getenv('TRAVIS_COMMIT_RANGE')
+if not travis_commit_range:
+    print("Travis commit range is empty, exiting...")
+    sys.exit(0)
+
+try:
+    result = subprocess.check_output(['git', 'diff', '--no-commit-id', '--name-status', '-r', travis_commit_range])
+except Exception as e:
+    print(e.output)
+    raise e
+
+files_added = result.decode('utf-8').splitlines()
+print(files_added)
+subdir_name = ""
+    
+for file_added in files_added:
+    file_added = file_added.split()
+
+    # Exclude certain files from some checks
+    if file_added[1].startswith("scripts/") or file_added[1] in ['README.md', '.travis.yml', '.gitattributes']:
+        print("Warning: modified non-gitian file", file_added[1])
+        continue
+
+    # Check that files are only added, not modified or deleted
+    if file_added[0] != 'A':
+        print("Error: modified or removed existing file:", file_added[1])
+        sys.exit(1)
```

Comment:
Someone might want to submit a different attestation, overwriting their previous one, for example when they had a problem in their build environment. But maybe that is rare enough we can ignore travis in that case - after all, this is only run on PRs and not on the branch.